affinity/cpp-20/d0796r1.md

```diff
@@ -165,94 +206,163 @@ There are some additional challenges which we have been investigating but are no
+    size_t concurrency() const noexcept;
 
-*Returns:*
+*Returns:* The collective concurrency available to this resource. More pecifically, the number of *threads of execution* collectively available to this `execution_resource` and any resources which are *members of*, recursively.
 
-      size_t partition_size() const noexcept;
+    std::vector<resource> resources() const noexcept;
 
-*Returns:*
+*Returns:* All `execution_resource`s which are *members of* this resource.
 
-      const execution_resource &partition(size_t i) const noexcept;
+    const execution_resource &member_of() const noexcept;
 
-*Returns:*
+*Returns:* The `execution_resource` which this resource is a *member of*.
 
-      const execution_resource &member_of() const noexcept;
+    std::string name() const noexcept;
 
-*Returns:*
+*Returns:* An implementation defined string.
 
-      std::string name() const noexcept;
+    bool can_place_memory() const noexcept;
 
-*Returns:*
+*Returns:* If this resource is capable of allocating memory with affinity, 'true'.
 
-      bool can_place_memory() const noexcept;
+    bool can_place_agent() const noexcept;
 
-*Returns:*
+*Returns:* If this resource is capable of execute with affinity, 'true'.
 
-      bool can_place_agent() const noexcept;
+## Class `execution_context`
 
-*Returns:*
+The `execution_context` class provides an abstraction for managing a number of light weight execution agents executing work on an `execution_resource` and any `execution_resource`s encapsulated by it. The `execution_resource` which an `execution_context` encapsulates is refered to as the *contained resource*.
 
-## Class `execution_context`
+### `execution_context` types
+
+    using executor_type = see-below;
+
+*Requires:* `executor_type` is an implementation defined class which satifies the general executor requires, as specified by P0443r5.
 
-The `execution_context` class provides an abstraction for managing a number of light weight execution agents executing work on an `execution_resource` and any `execution_resource`s encapsulated by it.
+    using pmr_memory_resource_type = see-below;
 
-### `execution_context` member aliases
+*Requires:* `pmr_memory_resource_type` is an implementation defined class which inherits from `std::pmr::memory_resource`.
 
-	using executor_type = __unspecfied__;
+    using allocator_type = see-below;
 
-*Requires:*
+*Requires:* `allocator_type` is an implementation defined allocator class.
 
 ### `execution_context` constructors
 
-	template <typename ExecutionResource>
-	execution_context(ExecutionResource &&execResource);
+    execution_context(const execution_resource &) noexcept;
 
-### `execution_context` destructor
+*Effects:* Constructs an `execution_context` with the provided resource as the *contained resource*.
 
+### `execution_context` destructor
+    
     ~execution_context();
 
+*Effects:* May or may not block to wait any work bveing executed on the *contained resource*.
```

Comment:
bveing -> being

Reply:
this should be clarified, destructor should wait, otherwise we risk leaking resources after main finishes and messing up with dynamic libraries

Reply:
I wasn't sure how we should define this but I've left this deliberately vague for now as P0737r0 defines properties which specify on destruction behavior, however, these are still being discussed so I didn't want to include them just yet.